dev/devicelab/lib/tasks/android_choreographer_do_frame_test.dart

```diff
@@ -0,0 +1,193 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../framework/framework.dart';
+import '../framework/task_result.dart';
+import '../framework/utils.dart';
+
+const List<String> kSentinelStr = <String>[
+  '==== sentinel #1 ====',
+  '==== sentinel #2 ====',
+  '==== sentinel #3 ====',
+];
+
+// Regression test for https://github.com/flutter/flutter/issues/98973
+// This test ensures that Choreographer#doFrame finishes during application startup.
+// This test failes if the application hangs during this period.
+// https://ui.perfetto.dev/#!/?s=da6628c3a92456ae8fa3f345d0186e781da77e90fc8a64d073e9fee11d1e65
+TaskFunction androidChoreographerDoFrameTest({
+  String? deviceIdOverride,
+  Map<String, String>? environment,
+}) {
+  final Directory tempDir = Directory.systemTemp
+      .createTempSync('flutter_devicelab_android_surface_recreation.');
+  return () async {
+    try {
+      section('Create app');
+      await inDirectory(tempDir, () async {
+        await flutter(
+          'create',
+          options: <String>[
+            '--platforms',
+            'android',
+            'app',
+          ],
+          environment: environment,
+        );
+      });
+
+      final File mainDart = File(path.join(
+        tempDir.absolute.path,
+        'app',
+        'lib',
+        'main.dart',
+      ));
+      if (!mainDart.existsSync()) {
+        return TaskResult.failure('${mainDart.path} does not exist');
+      }
+
+      section('Patch lib/main.dart');
+      await mainDart.writeAsString('''
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  print('${kSentinelStr[0]}');
+  await SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersiveSticky);
+
+  print('${kSentinelStr[1]}');
+  await SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersiveSticky);
```

Comment:
What does the second call to `setEnabledSystemUIMode(immersiveSticky)` do?

Reply:
it should do nothing, but this future won't resolve if the main thread is blocked. I can add a comment to clarify.